src/m365/planner/commands/roster/roster-member-remove.spec.ts

```diff
@@ -0,0 +1,311 @@
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if userId is not a valid guid.', async () => {
+    const actual = await command.validate({
+      options: {
+        rosterId: validRosterId,
+        userId: 'Invalid GUID'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('fails validation if userName is not a valid userName.', async () => {
+    const actual = await command.validate({
+      options: {
+        rosterId: validRosterId,
+        userName: 'John Doe'
+      }
+    }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (id)', async () => {
+    const actual = await command.validate({ options: { rosterId: validRosterId, userId: validUserId } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('passes validation if required options specified (name)', async () => {
+    const actual = await command.validate({ options: { rosterId: validRosterId, userName: validUserName } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before removing the specified roster member when confirm option not passed', async () => {
+    await command.action(logger, {
+      options: {
+        rosterId: validRosterId,
+        userid: validUserId
+      }
+    });
+    let promptIssued = false;
+
+    if (promptOptions && promptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('prompts before removing the last roster member when confirm option not passed', async () => {
+    let secondPromptOptions: any;
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async (options: any) => {
+      if (options.message === `Are you sure you want to remove member '${validUserId}'?`) {
+        return (
+          { continue: true }
+        );
+      }
+      else {
+        secondPromptOptions = options;
+        return (
+          { continue: false }
+        );
+      }
+    });
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if (opts.url === `https://graph.microsoft.com/beta/planner/rosters/${validRosterId}/members?$select=Id`) {
+        return (singleRosterMemberResponse);
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, {
+      options: {
+        rosterId: validRosterId,
+        userId: validUserId
+      }
+    });
+
+    let promptIssued = false;
+
+    if (secondPromptOptions && secondPromptOptions.type === 'confirm') {
+      promptIssued = true;
+    }
+
+    assert(promptIssued);
+  });
+
+  it('aborts removing the specified roster member when confirm option not passed and prompt not confirmed', async () => {
+    const deleteSpy = sinon.spy(request, 'delete');
+
+    sinonUtil.restore(Cli.prompt);
+    sinon.stub(Cli, 'prompt').callsFake(async () => (
```

Comment:
This stub can be removed because we are already cancelling all prompts in the beforeEach function.